sql/core/src/test/scala/org/apache/spark/sql/connector/SupportsCatalogOptionsSuite.scala

```diff
@@ -322,6 +323,12 @@ class SupportsCatalogOptionsSuite extends QueryTest with SharedSparkSession with
         timestamp = Some("2019-01-29 00:37:58")), df3.toDF())
       checkAnswer(load("t", Some(catalogName), version = None,
         timestamp = Some("2021-01-29 00:37:58")), df4.toDF())
+
+      // load with timestamp in number format
+      checkAnswer(load("t", Some(catalogName), version = None,
+        timestamp = Some(MICROSECONDS.toSeconds(ts1).toString)), df3.toDF())
```

Comment:
so this is not a bug fix but a new feature? I don't think it was designed to support microsecond as the timestamp string.

Reply:
My thought was considering seconds as the timestamp works in Spark SQL, it should work in data frame path also. 

> I don't think it was designed to support microsecond as the timestamp string

I now see, spark-sql AST builder un-intentionally creates a literal expression of longType in this scenario, which fits in the code flow path and works when we create a TimeTravelSpec and trigger the casting. Where as the data-frame code flow always forced us by design to return it in string format (implying date format should only given). 

Should I update the jira / pr description ?

Reply:
I think this over: do we need to support the microsecond? Seems to me it doesn't make sense for user to specify a long value in the query
```
SELECT * FROM table TIMESTAMP AS OF '1616574866666'
```

Reply:
For timestamps, only date or timestamp strings should be accepted. I don't think we should accept a long value.

Reply:
> I think this over: do we need to support the microsecond? Seems to me it doesn't make sense for user to specify a long value in the query 

+1 on this there is also one risk in supporting this as well, we need to specify the time always in **seconds** precision only, other wise it can return timestamp in wrong precision, to the catalog and hence making the query fail . [CodePointer](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L743) 


>  For timestamps, only date or timestamp strings should be accepted. I don't think we should accept a long value.

Are you suggesting we make changes in the grammar for not supporting it in the SQL ?

I raised this change considering : 
1. SQL was supporting this and df options were not (apologies, i thought it was kind of parity gap)
2. Iceberg supports time travel via custom options in data frame, such as [`as-of-timestamp`](https://iceberg.apache.org/docs/latest/spark-queries/#time-travel) which excepts the value of option as milliseconds value, since now spark supports this via SupportCatalogOptions, i thought we can use this.

Happy to make the changes accordingly, or even close it if we think it's not an issue from df reader perspective.

Reply:
> spark-sql AST builder un-intentionally creates a literal expression of longType in this scenario

I see. This is indeed un-intentional... But since it's already supported, we can't remove it now. Then it makes sense to make SQL and dataframe API consistent. @singhpk234 can you update the PR title and description? And also add SQL tests for this case.

Reply:
ACK